### PR TITLE
COMMERCE-9805 - It's possible to insert via API negative numeric values on product subscription fields

### PR DESCRIPTION
**Pull request by victorcdp**
Related Issue: [COMMERCE-9805](https://issues.liferay.com/browse/COMMERCE-9805)

The issue was that there was no backend validation for the input numbers.
Proposed fix is to add validation at the addCPDefinition and updateSubscriptionInfo methods.

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
@victorcdp , please address SF failure.

**Comment by victorcdp** [comment]
I don't even know how that extra space was there, since this was a generated file. Should work now.

**Comment by victorcdp** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -205,6 +206,13 @@ public CPDefinition addCPDefinition(
 
 		// Commerce product definition
 
+		if ((subscriptionLength < 1) || (maxSubscriptionCycles < 0) ||
+			(deliverySubscriptionLength < 1) ||
+			(deliveryMaxSubscriptionCycles < 0)) {
+
+			throw new CPSubscriptionDateException();
```
Comment (brianikim) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java]: @victorcdp , I want the lengths and subscriptionCycles to be separate checks with their own exception messages.

Ex.
```
			throw new CPDefinitionExpirationDateException(
				"Enter subscription length greater than or equal to 1");
```
```
			throw new CPDefinitionExpirationDateException(
				"Enter subscription cycles greater than or equal to 0");
```

Comment (victorcdp) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java]: Should be good now. We have to run sf again right?

Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java]: Looks great. Yes, always re-run sf when you make changes.

**Comment by victorcdp** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 2]
```diff
@@ -205,6 +206,18 @@ public CPDefinition addCPDefinition(
 
 		// Commerce product definition
 
+		if ((subscriptionLength < 1) || (deliverySubscriptionLength < 1)) {
+			throw new CPSubscriptionDateException(
+				"Enter subscription length greater than or equal to 1");
```
Comment (brianikim) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java]: Upon second thought, let's change the message to following format:

`Subscription length must be greater than or equal to 1`

Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java]: Furthermore, I want you to create a new private method with name `isValidSubscriptionInfo` and do your condition checks there.

Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java]: @victorcdp , see my above comments for requested changes.

Comment (victorcdp) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java]: Done. Should be good as well

**Comment by victorcdp** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 3]
```diff
@@ -2981,6 +2990,24 @@ private Map<String, String> _getUniqueUrlTitles(
...
 	}
 
+	private void _isValidSubscriptionInfo(
+			int subscriptionLength, int deliverySubscriptionLength,
+			long maxSubscriptionCycles, long deliveryMaxSubscriptionCycles)
+		throws CPSubscriptionDateException {
+
+		if ((subscriptionLength < 1) || (deliverySubscriptionLength < 1)) {
+			throw new CPSubscriptionDateException(
+				"Enter subscription length greater than or equal to 1");
```
Comment (brianikim) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java]: @victorcdp , alter the messages as well to the requested format:

`Subscription length must be greater than or equal to 1`

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
Good work @victorcdp

**Pull request closed without merging**